esphome/components/addressable_light_digital/digital_display.cpp

```diff
@@ -0,0 +1,160 @@
+#include "digital_display.h"
+#include "ascii_to_raw.h"
+#include "esphome/core/log.h"
+#include <cstdio>
+#include <cstring>
+
+namespace esphome {
+namespace addressable_light_digital {
+
+static const char *const TAG = "addressable_light_digital.display";
+
+light::LightTraits DigitalDisplay::get_traits() { return this->internal_light_output_->get_traits(); }
+
+void DigitalDisplay::setup_state(light::LightState *state) {
+  ESP_LOGD(TAG, "Setup state");
+  this->external_light_state_ = state;
+}
+
+void DigitalDisplay::write_state(light::LightState *state) {
+  ESP_LOGD(TAG, "Writing state");
+  this->internal_light_output_->update_state(state);
+  this->display_();
+}
+
+void DigitalDisplay::dump_config() {
+  ESP_LOGCONFIG(TAG,
+                "Addressable Light Digital:\n"
+                "  Update interval (ms): %d\n"
+                "  Number of LEDs: %d\n"
+                "  LED MAP: %s\n"
+                "  Max number of characters: %d\n"
+                "  Reverse: %s",
+                this->get_update_interval(), this->num_leds_, (this->led_map_).c_str(), this->max_characters_,
+                this->reverse_ ? "True" : "False");
```

Comment:
We have a macro for this =)
```suggestion
                TRUEFALSE(this->reverse_));
```

Reply:
Good suggestion. I might have missed it a couple of years ago.